lib/ansible/plugins/connection/docker.py

```diff
@@ -213,7 +213,7 @@ def exec_command(self, cmd, in_data=None, sudoable=False):
         local_cmd = [to_bytes(i, errors='surrogate_or_strict') for i in local_cmd]
         p = subprocess.Popen(local_cmd, shell=False, stdin=subprocess.PIPE,
                              stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-
+        p.wait()
```

Comment:
this is not recommended since we are using pipes, this can cause blocking if the output to the pipes exceeds the buffer size.

in any case the 'wait' is implied in the `p.communicate` below.

It seems that a sleep would allow the 'double execution' that is happening in the background and not block/break the communicate + pipe usage below. Still not a great fix but I would need to look closer into what is actually going on to come up with better.